### PR TITLE
Add publish status to get all reports endpoint

### PR DESCRIPTION
**Pull request by littledice**
First https://github.com/vimc/montagu-webmodels/pull/44

### CODE REVIEW

File: src/app/src/test/kotlin/org/vaccineimpact/reporting_api/tests/database_tests/OrderlyTests.kt [hunk 1]
```diff
@@ -35,10 +35,12 @@ class OrderlyTests : DatabaseTests()
         assertThat(results[0].name).isEqualTo("test")
         assertThat(results[0].displayName).isEqualTo("display name test")
         assertThat(results[0].latestVersion).isEqualTo("vz")
+        assertThat(results[0].published).isTrue()
 
         assertThat(results[1].name).isEqualTo("test2")
         assertThat(results[1].displayName).isEqualTo("display name test2")
         assertThat(results[1].latestVersion).isEqualTo("vb")
+        assertThat(results[0].published).isTrue()
```
Comment (MartinEden) [inline src/app/src/test/kotlin/org/vaccineimpact/reporting_api/tests/database_tests/OrderlyTests.kt]: This should be `assertThat(results[1].published).isFalse()`

File: src/app/src/test/resources/spec/Report.schema.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "properties": {
     "name": {"type" :"string"},
     "display_name": {"type": [ "string", "null"] },
-    "latest_version": { "type": "string" }
+    "latest_version": { "type": "string" },
+    "published": { "type": "boolean" }
   },
-  "required": ["name", "latest_version", "display_name"]
+  "required": ["name", "latest_version", "display_name", "published"]
 }
```
Comment (MartinEden) [inline src/app/src/test/resources/spec/Report.schema.json]: Is there a blackbox test that makes use of this schema file?

Comment (littledice) [reply src/app/src/test/resources/spec/Report.schema.json]: This schema is just used by the other schema file, `Reports.schema.json` and there is a test that validates against that (the first test in `ReportTests`)

File: src/app/src/test/kotlin/org/vaccineimpact/reporting_api/tests/database_tests/OrderlyTests.kt [hunk 3]
```diff
@@ -35,10 +35,12 @@ class OrderlyTests : DatabaseTests()
         assertThat(results[0].name).isEqualTo("test")
         assertThat(results[0].displayName).isEqualTo("display name test")
         assertThat(results[0].latestVersion).isEqualTo("vz")
+        assertThat(results[0].published).isTrue()
 
         assertThat(results[1].name).isEqualTo("test2")
         assertThat(results[1].displayName).isEqualTo("display name test2")
         assertThat(results[1].latestVersion).isEqualTo("vb")
+        assertThat(results[1].published).isTrue()
```
Comment (littledice) [inline src/app/src/test/kotlin/org/vaccineimpact/reporting_api/tests/database_tests/OrderlyTests.kt]: Ok, this is the right logic now!

**Pull request merged**